### ISSUE TITLE
Database SQL export only processes visible tables when using a filter

### ISSUE THREAD

**Issue by webloft** [HeidiSQL/HeidiSQL#1983]
**Preconditions**
* HeidiSQL version: 12.8.0.6908
* Database type and version: MySQL
* OS: Windows

**Describe the bug**

Database SQL export only processes last visible tables when using a filter. The normal behaviour of such a filter should made it easy to select a table by a filter but also respect previously selected tables and not ignore them.

**To Reproduce**

1. Untick all tables in database export dialog
2. Set a table filter and select some tables
3. Change the filter and select some other tables
4. Start export

**Comment by vohuynhminhtan** [comment]
Please add a 'Check Filtered Tables' option to allow for selecting all filtered results at once. Manually selecting dozens of filtered tables is a nightmare. Because of this enhancement, quick selection of filtered tables is no longer possible.

**Comment by ansgarbecker** [comment]
Makes sense, thanks for the hint. Next builds change the "Check all" action to "Change all visible" when a filter is active:

<img width="353" height="185" alt="Image" src="https://github.com/user-attachments/assets/3a17c0b5-8f85-4b23-af7a-9517d5357466" />
